### PR TITLE
AP-3083: Replace secure header gem

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3083)

Replace the secure_header gem with Rail's built-in version

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: config/initializers/content_security_policy.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                       "https://www.google-analytics.com",
+                       "https://*.justice.gov.uk"
+  else
+    policy.script_src :self,
+                      "https://www.google-analytics.com",
+                      "https://www.googletagmanager.com"
+    policy.connect_src :self,
+                       "https://www.google-analytics.com",
+                       "https://*.justice.gov.uk",
+                       "http://localhost:*"
```
Comment (cpjmcquillan) [inline config/initializers/content_security_policy.rb:28]: Is this supposed to be added in development instead?

The previous config had 

```ruby
connect_src = %w['self' https://www.google-analytics.com https://*.justice.gov.uk] 
connect_src << "http://localhost:*" if Rails.env.development?
```

**Review by cpjmcquillan — APPROVED** [review]

**Review by jsugarman — COMMENTED** [review]
getting a 

```
Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' 'unsafe-inline' 'nonce-Q5vYJvDZtUYhntemn29lOg=='". Note that 'unsafe-inline' is ignored if either a hash or nonce value is present in the source list.
```

Screenshot 2022-12-12 at 16 57 54


...on the chrome JS console.

I have had this happen to me before so 👀

**Comment by jsugarman** [comment]
> getting a
> 
> ```
> Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' 'unsafe-inline' 'nonce-Q5vYJvDZtUYhntemn29lOg=='". Note that 'unsafe-inline' is ignored if either a hash or nonce value is present in the source list.
> ```
>...
> I have had this happen to me before so 👀

This looks like it fixes the problem BUT by explicitly not using the nonce for the `style-src` directive  - which `main` and `production` do not - nor for `script-src` in development. 

This is required because `unsafe-inline` is ignored if a nonce is specified, and rails CSP defaults to adding a nonce to `style-src` directive and `script-src` directive.
 
```
# set nonce only for directives you _do not want_ :unsafe_inline specified for
# https://github.com/rails/rails/pull/36534/commits/09d55b302266cf002a4b307f8d37a105d2838a18
if Rails.env.development?
  Rails.application.config.content_security_policy_nonce_directives = []
else
  Rails.application.config.content_security_policy_nonce_directives = %w(script-src)
end

```

**Review by jsugarman — APPROVED** [review]
Looks good. Have not UAT'd as yet though

**Comment by colinbruce** [comment]
I wasn't going to smash merge this. I will ask Mike to regression test everything including Truelayer :+1:

**Pull request merged**